execution/evm/native.go

```diff
@@ -140,3 +151,139 @@ func identityFunc(state Interface, caller crypto.Address, input []byte, gas *uin
 	// Return identity
 	return input, nil
 }
+
+// from go-ethereum/blob/master/core/vm/contracts.go
+var (
+	big1      = big.NewInt(1)
+	big4      = big.NewInt(4)
+	big8      = big.NewInt(8)
+	big16     = big.NewInt(16)
+	big32     = big.NewInt(32)
+	big64     = big.NewInt(64)
+	big96     = big.NewInt(96)
+	big480    = big.NewInt(480)
+	big1024   = big.NewInt(1024)
+	big3072   = big.NewInt(3072)
+	big199680 = big.NewInt(199680)
+)
+
+const (
+	MaxUint64 = 1<<64 - 1
+
+	ModExpQuadCoeffDiv uint64 = 20 // Divisor for the quadratic particle of the big int modular exponentiation
+)
+
+func bigModExp(state Interface, caller crypto.Address, input []byte, gas *uint64, logger *logging.Logger) (output []byte, err error) {
+	//Deduct gas
+	gasRequired := bigModExpGasRequire(input)
+	if *gas < gasRequired {
+		return nil, errors.ErrorCodeInsufficientGas
+	}
+	*gas -= gasRequired
+
+	var (
+		baseLen = new(big.Int).SetBytes(getData(input, 0, 32)).Uint64()
+		expLen  = new(big.Int).SetBytes(getData(input, 32, 32)).Uint64()
+		modLen  = new(big.Int).SetBytes(getData(input, 64, 32)).Uint64()
+	)
+	if len(input) > 96 {
+		input = input[96:]
+	} else {
+		input = input[:0]
+	}
+	// Handle a special case when both the base and mod length is zero
+	if baseLen == 0 && modLen == 0 {
+		return []byte{}, nil
+	}
+	// Retrieve the operands and execute the exponentiation
+	var (
+		base = new(big.Int).SetBytes(getData(input, 0, baseLen))
+		exp  = new(big.Int).SetBytes(getData(input, baseLen, expLen))
+		mod  = new(big.Int).SetBytes(getData(input, baseLen+expLen, modLen))
+	)
+	if mod.BitLen() == 0 {
+		// Modulo 0 is undefined, return zero
+		return LeftPadBytes([]byte{}, int(modLen)), nil
+	}
+	return LeftPadBytes(base.Exp(base, exp, mod).Bytes(), int(modLen)), nil
+
+}
+
+// RequiredGas returns the gas required to execute the pre-compiled contract.
+func bigModExpGasRequire(input []byte) uint64 {
```

Comment:
For our purposes I think we can just elide this complex gas calculation. Gas is something we need to revisit at some stage, but in Burrow right now all gas does is bound computation. A unitary gas cost for each operation is sufficient for this.